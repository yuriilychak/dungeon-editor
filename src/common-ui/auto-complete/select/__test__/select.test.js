import React from "react";
import { createMount } from "@material-ui/core/test-utils";

import Select from "../select";

describe("Select test",()=> {
    const defaultProps = {
        getMenuProps: jest.fn(),
        getItemProps: jest.fn(),
        suggestions: [
            {
                item: "test"
            }
        ],
        inputValue: "test",
        isOpen: true,
        showEmpty: true,
        highlightedIndex: 0
    };

    it("Default snapshot", () => {
        const wrapper = createMount()(
            <Select {...defaultProps} />
        );

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Snapshot closed", () => {
        const wrapper = createMount()(
            <Select {...defaultProps} isOpen={false}/>
        );

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Snapshot without show empty", () => {
        const wrapper = createMount()(
            <Select {...defaultProps} showEmpty={false}/>
        );

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Snapshot with empty input", () => {
        const wrapper = createMount()(
            <Select {...defaultProps} inputValue={""}/>
        );

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Snapshot with empty input and without show empty", () => {
        const wrapper = createMount()(
            <Select {...defaultProps} inputValue={""} showEmpty={false}/>
        );

        expect(wrapper.html()).toMatchSnapshot();
    });
});
