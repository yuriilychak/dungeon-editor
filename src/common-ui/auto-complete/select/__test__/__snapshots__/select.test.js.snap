// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select test Default snapshot 1`] = `"<div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-root-1\\"><div class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-29 makeStyles-menuItemDefault-31 Mui-selected MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">test<span class=\\"MuiTouchRipple-root\\"></span></div></div></div>"`;

exports[`Select test Snapshot closed 1`] = `"<div></div>"`;

exports[`Select test Snapshot with empty input 1`] = `"<div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-root-1\\"><div class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-29 makeStyles-menuItemDefault-31 Mui-selected MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">test<span class=\\"MuiTouchRipple-root\\"></span></div></div></div>"`;

exports[`Select test Snapshot with empty input and without show empty 1`] = `"<div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-root-1\\"></div></div>"`;

exports[`Select test Snapshot without show empty 1`] = `"<div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-root-1\\"><div class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-29 makeStyles-menuItemDefault-31 Mui-selected MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">test<span class=\\"MuiTouchRipple-root\\"></span></div></div></div>"`;
